src/blister.h

```diff
@@ -208,30 +208,107 @@ static inline UInt * BLOCKS_BLIST(Obj list)
-        (block) = ((block) & 0x5555555555555555L) + (((block) >> 1) & 0x5555555555555555L); \
-        (block) = ((block) & 0x3333333333333333L) + (((block) >> 2) & 0x3333333333333333L); \
-        (block) = ((block) + ((block) >>  4)) & 0x0f0f0f0f0f0f0f0fL;                        \
-        (block) = ((block) + ((block) >>  8));                                              \
-        (block) = ((block) + ((block) >> 16));                                              \
-        (block) = ((block) + ((block) >> 32)) & 0x00000000000000ffL; } while (0)            
-
+    block =
+        (block & 0x5555555555555555L) + ((block >> 1) & 0x5555555555555555L);
+    block =
+        (block & 0x3333333333333333L) + ((block >> 2) & 0x3333333333333333L);
+    block = (block + (block >> 4)) & 0x0f0f0f0f0f0f0f0fL;
+    block = (block + (block >> 8));
+    block = (block + (block >> 16));
+    block = (block + (block >> 32)) & 0x00000000000000ffL;
 #else
-
-#define COUNT_TRUES_BLOCK( block )                                        \
-        do {                                                              \
-        (block) = ((block) & 0x55555555) + (((block) >> 1) & 0x55555555); \
-        (block) = ((block) & 0x33333333) + (((block) >> 2) & 0x33333333); \
-        (block) = ((block) + ((block) >>  4)) & 0x0f0f0f0f;               \
-        (block) = ((block) + ((block) >>  8));                            \
-        (block) = ((block) + ((block) >> 16)) & 0x000000ff; } while (0)   
+    block = (block & 0x55555555) + ((block >> 1) & 0x55555555);
+    block = (block & 0x33333333) + ((block >> 2) & 0x33333333);
+    block = (block + (block >> 4)) & 0x0f0f0f0f;
+    block = (block + (block >> 8));
+    block = (block + (block >> 16)) & 0x000000ff;
+#endif
+    return block;
 #endif
+}
 
+/****************************************************************************
+**
+*F  COUNT_TRUES_BLOCKS( <ptr>, <nblocks> ) 
+**  
+**  'COUNT_TRUES_BLOCKS( <ptr>, <nblocks> )' returns the total number of 1
+**  bits in the array of UInt values starting at <ptr> and including a total
+**  of <nblocks> UInts. The only reason this function is really needed is
+**  that, owing to hardware bugs and compiler peculiarities current in 2017,
+**  (see http://danluu.com/assembly-intrinsics/ or
+**  https://stackoverflow.com/questions/25078285?) manually unrolling this
+**  loop makes the code substantially faster on almost all CPUS.
+** 
+**  This interacts strangely with the choice of algorithm for
+**  COUNT_TRUES_BLOCK above. Without the loop unrolling, not using the gcc
+**  builtin is sometimes faster, apparently because it allows the compiler
+**  to unroll the loop and then generate SSE or AVX code to process multiple
+**  words at once. With the loop unrolling the builtin is always faster, and
+**  will itself generate AVX code when compiling for suitable processors.
+**
+*T  monitor this situation periodically. 
```

Comment:
Could we perhaps this to something more greppable, like: `TODO: monitor ...`?

Reply:
Obviously yes, but there are a dozen or so TODOs marked this way in the kernel and a lot marked in the analagous way `#T` in the library. Might be better to fix all of them in another PR?

Reply:
Yes, as I wrote in my approval message, it's fine to fix these things later on.

(I actually thought @markuspf had merged this PR by now, but he is presenting datastructures, so I'll do it)